aict_tools/scripts/train_dxdy_regressor.py

```diff
@@ -0,0 +1,174 @@
+from ..configuration import AICTConfig
+from ..logging import setup_logging
+
+
+@click.command()
+@click.argument('configuration_path', type=click.Path(exists=True, dir_okay=False))
+@click.argument('signal_path', type=click.Path(exists=True, dir_okay=False))
+@click.argument('predictions_path', type=click.Path(exists=False, dir_okay=False))
+@click.argument('dxdy_model_path', type=click.Path(exists=False, dir_okay=False))
+@click.option('-k', '--key', help='HDF5 key for h5py hdf5', default='events')
+@click.option('-v', '--verbose', help='Verbose log output', is_flag=True)
+def main(configuration_path, signal_path, predictions_path, dxdy_model_path, key, verbose):
+    '''
+    Train two learners to be able to reconstruct the source position.
+    One regressor for dx and one regressor for dy.
+
+    Both pmml and pickle format are supported for the output.
+
+    CONFIGURATION_PATH: Path to the config yaml file
+
+    SIGNAL_PATH: Path to the signal data
+
+    PREDICTIONS_PATH : path to the file where the mc predictions are stored.
+
+    DXDY_MODEL_PATH: Path to save the dxdy model to.
+
+        Allowed extensions are .pkl and .pmml.
+        If extension is .pmml, then both pmml and pkl file will be saved
+    '''
+    log = setup_logging(verbose=verbose)
+
+    config = AICTConfig.from_yaml(configuration_path)
+    model_config = config.dxdy
+
+    np.random.seed(config.seed)
+
+    dxdy_regressor = model_config.dxdy_regressor
+
+    dxdy_regressor.random_state = config.seed
+
+    log.info('Loading data')
+    df = read_telescope_data(
+        signal_path, config,
+        model_config.columns_to_read_train,
+        feature_generation_config=model_config.feature_generation,
+        n_sample=model_config.n_signal
+    )
+    log.info('Total number of events: {}'.format(len(df)))
+
+    log.info(
+        'Using coordinate transformations for %s',
+        model_config.coordinate_transformation
+    )
+
+    df = convert_units(df, model_config)
+    source_x, source_y = horizontal_to_camera(df, model_config)
+    
+    log.info('Using projected disp: {}'.format(model_config.project_disp))
+    #df['true_dx'] = source_x
+    #df['true_dy'] = source_y
+
+    df['true_dx'] = df[model_config.cog_x_column] - source_x
+    df['true_dy'] = df[model_config.cog_y_column] - source_y
+
+    ####
+    df['delta_err'] = delta_error(df, model_config)
+    log.info('Events after delta cut : {}'.format(sum(delta_error(df, model_config) < 10)))      # < 10: delta never gets that large
+    df = df.query(f'delta_err < 10')
+    ####
+
+    # generate features if given in config
+    if model_config.feature_generation:
+        feature_generation(df, model_config.feature_generation, inplace=True)
+
+    df_train = convert_to_float32(df[model_config.features])
+    df_train.dropna(how='any', inplace=True)
+
+    log.info('Events after nan-dropping: {} '.format(len(df_train)))
+
+    target_dx = df['true_dx'].loc[df_train.index]
```

Comment:
`target_dxdy = df.loc[df_train.index, ['target_dx', 'target_dy']].to_numpy()` and you don't need the stacking.